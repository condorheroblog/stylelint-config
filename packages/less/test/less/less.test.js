import { readFileSync } from "node:fs";
import { join } from "node:path";
import stylelint from "stylelint";
import { beforeEach, describe, expect, test } from "vitest";

import basicConfig from "../..";

const config = { ...basicConfig, ignoreFiles: [] };
const validateLessPath = join(__dirname, "./index.less");
const validateLess = readFileSync(validateLessPath, "utf-8");

describe("validate LESS", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validateLessPath,
      configBasedir: __dirname,
    });
  });

  test("did not error", () => {
    return result.then(data => expect(data.errored).toBeFalsy());
  });

  test("flags no warnings", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(0));
  });
});

describe("LESS pieces to be tested", () => {
  test("index.less", () => {
    return expect(validateLess).toMatchSnapshot();
  });
});
