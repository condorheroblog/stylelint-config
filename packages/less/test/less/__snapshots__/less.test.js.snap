// Vitest Snapshot v1

exports[`LESS pieces to be tested > index.less 1`] = `
"@link-color: #428bca;

:root {
  --color: @link-color;
}

.bordered {
  border-top: dotted 1px black;
  border-bottom: solid 2px black;
}

#menu a {
  color: #111;
  .bordered();
}

.parent {
  color: @link-color;

  & .child {
    width: 22px;
    height: 33px;
    color: #abc321;
  }
}
"
`;
